### PR TITLE
Apply style guide to the specs

### PR DESCRIPTION
**Pull request by edthamm**
# Description

This PR updates some configuration files and fixes style issues in the specs.

This is the first of two related PR's the next one will be along in the coming days
and will do the same for everything in lib. I.e. assuming you want/like this kind
of work to be done there or at all.
I decided to start with the specs because they are often not as 'clever' as the production
code while providing a reasonably large codebase to discuss style issues.
Also, I did not want to change everything at once because refactoring tests and production
code at the same time usually ends up breaking something.

This PR is also meant to serve as a starting point for discussing the style this project
wants.

Updated configuration:

- Rubocop dependency upped to 0.54.0 == current version on hound
- Update rubocop settings files to match new version
- Set Style/StringLiterals to single-quote > Seems to be the current default
- Set Layout/DotPosition to leading > Seems to be the current default
- Set Metrics/LineLength to 125 > Not the default but less annoying 
- Set CI timeout to 45 seconds on JRuby

Used rubocops auto-correct as far as possible. 

## Type of change

- Code cleanup

## Notes to the reviewer

This is quite a huge PR.

The commits are prefaced with:

|Label|Description|
|-----|-----------|
|Settings| Change and adapt settings|
|Auto correct| Fix style violations using rubocop auto-correct|
|Correct| Fix style violation manually|
|Fix| Fix an error or failing test case after adaptations|

Depending on what is changed in the commit.

I noticed that rubocop 0.54.0 does not play nice with the configuration files on my setup so I made the configuration explicit in the Guardfile.

Also at one point, I thought that ruby < 2.3 is not supported anymore so I dropped
support for that. After I noticed my error I undid those changes.

# How Has This Been Tested?

- [x] Run all existing tests locally in ruby 2.2.9, 2.5.0 and jruby 9.1.15.0
- [x] Run all existing tests in CI
- [x] Run bundle exec rubocop -c .rubocop.yml --rails spec  

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Comment by edthamm** [comment]
Hi!

So two things:

This build timed out prior to starting work
https://travis-ci.org/guard/guard/jobs/417820539 
could you please restart it.

Hound seems to be using the setup from master for the check. Therefore there are a lot of errors. This is due to the nature of the change imo.

File: vendor/hound/config/style_guides/ruby.yml [hunk 1]
```diff
@@ -102,7 +101,7 @@ Style/StringLiterals:
   Description: Checks if uses of quotes match the configured preference.
   StyleGuide: https://github.com/bbatsov/ruby-style-guide#consistent-string-literals
   Enabled: true
-  EnforcedStyle: double_quotes
```
Comment (rymai) [inline vendor/hound/config/style_guides/ruby.yml]: I think we should keep the `double_quotes` here, there's no need to change quotes style in the entire codebase.

Comment (edthamm) [reply vendor/hound/config/style_guides/ruby.yml]: Fine. I'll change that back.
What is your opinion on DotPosition?

Comment (rymai) [reply vendor/hound/config/style_guides/ruby.yml]: > What is your opinion on DotPosition?

I like it. 👍

**Comment by rymai** [comment]
@edthamm Not sure why Hound is still pending... 🤔

**Comment by edthamm** [comment]
@rymai I assume their instance got killed before their run finished. I currently don't have access to my computer and I guess hound still has not added a rerun feature. So I suggest we wait until I have access again and I add a fake commit in order to restart the analysis. Any better idea's?

**Comment by edthamm** [comment]
@rymai 
I checked the failing build. Seems to me like a timeout issue.
Could you restart the failed build please?
Is this something that happens at a regular basis?

**Comment by rymai** [comment]
@edthamm Thanks for your contribution!

**Pull request merged**